lib/trento/application/usecases/clusters/adapters/wanda.ex

```diff
@@ -0,0 +1,53 @@
+defmodule Trento.Clusters.Wanda do
+  @moduledoc """
+  Wanda integration adapter
+  """
+
+  @behaviour Trento.Clusters.Gen
+
+  import Ecto.Query
+
+  require Logger
+
+  alias Trento.{
+    ClusterReadModel,
+    HostReadModel
+  }
+
+  alias Trento.Integration.Checks
+
+  alias Trento.Repo
+
+  @impl true
+  def request_checks_execution(cluster_id) do
+    ClusterReadModel
+    |> Repo.get(cluster_id)
+    |> maybe_request_checks_execution()
+  end
+
+  defp maybe_request_checks_execution(nil), do: {:error, :cluster_not_found}
+  defp maybe_request_checks_execution(%{selected_checks: []}), do: :ok
+
+  defp maybe_request_checks_execution(%{
+         id: cluster_id,
+         provider: provider,
+         selected_checks: selected_checks
+       }) do
+    hosts_data =
```

Comment:
What about moving this second query next to the first one? Or even doing only one query?
So we can separate querying for the needed data for the execution and maybe requesting an execution based on the queried data as we're doing in `maybe_request_checks_execution`

Reply:
And: what happens/should happen if `hosts_data` ends up in being an empty list? Should it skip requesting an execution to start, similarly to when we get an empty selection?

Reply:
It shouldn't be possible to have a cluster without any host (unless we have some bug in our code I guess hehe).

About the queries order. I did it this way because it looked the easiest. If we start joining the cluster and host tables, we would already have multiple rows, and this would make all pattern matching more complex and not so obvious. With the current code, I feel that everyone coming to the code can easily understand the logic in a simple glance.

Anyway, if you have some code suggestion that looks good enough we can give a try. I couldn't get antyhing much better than this XD

Reply:
I see, I tried myself and yes, it can become really fancy :sweat_smile: 

Fine with this :wink: